[SherChew27] iP

SherChew27 opened the pull request:
# Bellroy 🦉 🦉 🦉 🦉 
> "Due on Sunday, do on Sunday" - Harith Hakim Bin Mazlan

## Bellroy allows you to focus on the things that matter, by freeing up your mind from the small things. It's:

- Clean
- Simple
- ~~Lightweight~~ Featherweight

and best of all, it's **FREE**!

Here are some examples of commands that you can use:

1. todo
2. deadline
3. event
4. mark
5. unmark
6. list
7. bye

This is what Bellroy can do:
- [X] Track your tasks 👍 
- [ ] Send deadline reminders
- [ ] Share your tasks

[Here is the next update that is currently in progress.](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html)

Here's the `main` method: 
``` java
public static void main(String[] args) {
        new Bellroy("Bellroy.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

SherChew27 committed:
Add Rename, Greet, Exit

SherChew27 committed:
Add Echo

SherChew27 committed:
Add Add, List

SherChew27 committed:
Add Mark as Done

SherChew27 committed:
Add ToDos, Events, Deadlines

SherChew27 committed:
Add Automated Text UI Testing

SherChew27 committed:
Add Exception Handling

SherChew27 committed:
Add Delete

SherChew27 committed:
Add writing to output file

SherChew27 committed:
Add reading from input file

SherChew27 committed:
Add Dates and Times

SherChew27 committed:
Add MoreOOP

SherChew27 committed:
Add Bellroy Packaging

SherChew27 committed:
Merge branch 'add-gradle-support' of https://github.com/SherChew27/ip

SherChew27 committed:
Add JUnit Test

SherChew27 committed:
Add JAR Creation

SherChew27 committed:
Add JavaDoc Comments

SherChew27 committed:
Add find

SherChew27 committed:
Merge branch 'branch-Level-9'

iamdiluxedbutcooler commented on src/test/java/Bellroy/BellroyTest.java:
```diff
@@ -0,0 +1,34 @@
+            file.delete();
+        }
+    }
+
```
The tests should also include JavaDoc comments too to clarify the purpose and scope of each test case.

iamdiluxedbutcooler commented on src/main/java/Bellroy/Ui.java:
```diff
@@ -0,0 +1,112 @@
+        System.out.println("____________________________________________________________\n" +
+                " Hello! I'm Bellroy\n" +
+                " What can I do for you?\n" +
+                "____________________________________________________________\n");
```
To improve the object-oriented design, consider refactoring the code by creating a private method that handles the printing of the horizontal line and message. This method could be structured to take a message as an argument and encapsulate the functionality of printing the message flanked by horizontal lines and would enhance modularity and reduces redundancy.

iamdiluxedbutcooler reviewed (commented):
WOW😯

M-kew commented on src/main/java/Bellroy/Ui.java:
```diff
@@ -0,0 +1,112 @@
+     * prints the initial message to the user.
+     */
+    public void printWelcomeMessage() {
+        System.out.println("____________________________________________________________\n" +
```
Perhaps these repeated lines of System.out.println("_________") can be put into a function called printLine()? Would be easier if you have to make modifications to it :)

M-kew commented on src/main/java/Bellroy/TaskList.java:
```diff
@@ -0,0 +1,78 @@
+
+    public List<Task> findTask(String keyword) {
+        List<Task> matchingTask = new ArrayList<>();
+        for (Task task: taskList) {
```
Would be good if you can add 1 indentation between task and : (based on coding standard).

M-kew commented on src/main/java/Bellroy/Storage.java:
```diff
@@ -0,0 +1,83 @@
+                        taskList.add(tempToDo);
+                        break;
+                    case ('D'):
+                        String[] deadlineParts = remainder.split(" \\(by: ");
```
Perhaps you can also add some inline comments on what this chunk of code snippet does as it would look overwhelming at first glance for readers :)

M-kew commented on src/main/java/Bellroy/Todo.java:
```diff
@@ -0,0 +1,18 @@
+
+    @Override
+    public String toString() {
+        String message = "[" + this.type + "]" + "[" + this.getStatusIcon() + "] " + this.description;
```
I used StringBuilder() for String that has multiple concatenation operation as it would be more optimised. Maybe you can also try using it? Do let me know your thoughts! :)

M-kew commented on src/main/java/Bellroy/Parser.java:
```diff
@@ -0,0 +1,75 @@
+        String type = input[0].split(" ")[0].toLowerCase();
+        String description = input[0].substring(type.length());
+        try {
+            switch (type) {
```
Good use of switch statement! Code looks clean 👍

M-kew commented on src/main/java/Bellroy/Bellroy.java:
```diff
@@ -0,0 +1,46 @@
+     */
+    public void run() {
+        ui.printWelcomeMessage();
+        boolean isRunning = true;
```
Good use of prefixes for boolean variable naming :)

M-kew commented on src/main/java/Bellroy/Task.java:
```diff
@@ -0,0 +1,28 @@
+package Bellroy;
```
Would be good if can create a separate package. For example, all tasks-related classes such as Event, Deadline, and Todo can be under Bellroy.Task package.

M-kew commented on src/main/java/Bellroy/Storage.java:
```diff
@@ -0,0 +1,83 @@
+        return taskList;
+    }
+
+    public void save(TaskList taskList) throws IOException{
```
Can add an identation between IOException and {.

M-kew commented on src/main/java/Bellroy/Ui.java:
```diff
@@ -0,0 +1,112 @@
+    }
+
+    /**
+     * prints out message to show that the task is deleted
```
Javadocs are well-written. Simple and easy to understand. 👍

M-kew reviewed (commented):
LGTM. Just some minor changes that has to be made. :)

siriousguy commented on src/main/java/Bellroy/Bellroy.java:
```diff
@@ -0,0 +1,46 @@
+
+import java.io.IOException;
+
+/**
+ * Represents an instance of the Bellroy Chatbot.
+ */
+
+public class Bellroy {
```
Minor coding standard adherence that can be done here by removing line 8:
"No blank line between the documentation block and the method/class."

siriousguy commented on src/main/java/Bellroy/Task.java:
```diff
@@ -0,0 +1,28 @@
+ * Abstract Task Class
+ */
+abstract class Task {
+    protected String description;
+    protected boolean isDone;
+    protected String type;
```
Good use of protected here!

siriousguy commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,3 @@
+todo swim
+Bellroy.deadline homework /by 2359
+event birthday party /from 4pm /to 6pm
```
Good use of test inputs!

siriousguy commented on src/main/java/Bellroy/Todo.java:
```diff
@@ -0,0 +1,18 @@
+/**
+ * Encapsulates a Todo Task
+ */
+public class Todo extends Task {
+
+    public Todo(String description) {
```
a very minor change, but to do should either be two words or written to-do, so a "ToDo" class is more aligned.

(Sorry, the other reviewer did an excellent job so I'm really nitpicking here)

siriousguy commented on src/main/java/Bellroy/Ui.java:
```diff
@@ -0,0 +1,112 @@
+     * @param task to be deleted
+     * @param size number of tasks remaining after deletion
+     */
+    public void taskDeleted(Task task, int size) {
+        System.out.printf("    ____________________________________________________________\n" +
+                "     Got it. I've removed this task:\n" +
+                "       " + task + "\n" +
+                "     Now you have %d tasks in the list.\n" +
+                "    ____________________________________________________________\n", size);
+    }
```
Might want to consider a verb form for this method such as "showDeleted"
- Names representing methods must be verbs and written in camelCase.

siriousguy reviewed (commented):
LGTM! To be nitpicking, can consider naming the methods in a verb form. The other reviewer has mentioned most things, but to add on, the programme's logic is straightforward, and the Javadocs have been written in a very accessible manner.

SherChew27 committed:
Edit method names

SherChew27 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Bellroy/Ui.java

SherChew27 committed:
Add GUI

SherChew27 committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	src/main/java/Bellroy/Ui.java
#	src/main/java/Bellroy/parser/Parser.java

SherChew27 committed:
no message

SherChew27 committed:
Add Assertions

SherChew27 committed:
Merge pull request #1 from SherChew27/branch-A-Assertions

Add Assertions

SherChew27 committed:
Implement SLAP in Storage class.

Instead of processing the input within the loadTask method, the processing is abstracted into individual methods.

The additional methods abstract out the processing of the string into its relevant components so that the loadTask method's logic is mainly to add the tasks to the tasklist.
This makes it easier to understand the logic of the loadTask method without knowing how it is implemented.

SherChew27 committed:
Merge pull request #2 from SherChew27/branch-A-CodeQuality

Implement SLAP in Storage class.

SherChew27 committed:
Add Client extension feature

SherChew27 committed:
Fixed Welcome Message for GUI

SherChew27 committed:
Change method names to Bellroy instead of Duke

SherChew27 committed:
Remove variable sendButton that is no longer used

SherChew27 committed:
Change Ui class methods to be static

SherChew27 committed:
Add default case for Parser and Storage

SherChew27 committed:
Update Parser class to make static call instead of method call

SherChew27 committed:
Add Personality to Bellroy

SherChew27 committed:
Add Title Bar

SherChew27 committed:
Add Ui.png

SherChew27 committed:
Add User Guide

SherChew27 committed:
Add more javadoc comments

SherChew27 committed:
Edit the error catching for parser class to be more specific

SherChew27 committed:
Add error catching and handling for wrong deadline format, JUnit test to test that

SherChew27 committed:
Add Error Handling for wrong input for deadline and event tasks

SherChew27 committed:
Change format of Deadline Task toString

SherChew27 committed:
Add Message for empty tasklist

SherChew27 committed:
Update Parser class with SLAP

SherChew27 committed:
Edit Parser class to slap for task creation

SherChew27 committed:
Edit Parser class to fully SLAP

SherChew27 committed:
Rename package to adhere to naming convention

SherChew27 committed:
Update README.md

SherChew27 committed:
Update README.md

SherChew27 committed:
Update README.md

SherChew27 committed:
Update README.md

SherChew27 committed:
Update README.md

SherChew27 committed:
Update README.md
